File: clases/26.2-contador-cliente.js
const baseUrl = "http://localhost:5001/contador";

async function leerContador() {
  const response = await fetch(baseUrl);
  return await response.text();
}

async function enviarPut() {
  const response = await fetch(baseUrl, { method: "PUT" });
  return await response.text();
}

async function enviarDelete() {
  const response = await fetch(baseUrl, { method: "DELETE" });
  return await response.text();
}

async function main() {
  const inicial = await leerContador();
  console.log(`Estado inicial: ${inicial}`);

  console.log(`Incrementar 1: ${await enviarPut()}`);
  console.log(`Incrementar 2: ${await enviarPut()}`);

  const despuesDeIncrementar = await leerContador();
  console.log(`Después de incrementar 2 veces: ${despuesDeIncrementar}`);

  console.log(`Borrar contador: ${await enviarDelete()}`);

  const final = await leerContador();
  console.log(`Estado final: ${final}`);
}

main();

File: clases/26.2-contador-cliente.py
import json
import urllib.request

base_url = "http://localhost:5001/contador"


def leer_contador():
    with urllib.request.urlopen(base_url) as response:
        return json.load(response)["contador"]


def enviar_put():
    request = urllib.request.Request(base_url, method="PUT")
    with urllib.request.urlopen(request):
        pass


def enviar_delete():
    request = urllib.request.Request(base_url, method="DELETE")
    with urllib.request.urlopen(request):
        pass


inicial = leer_contador()
print(f"Estado inicial: {inicial}")

enviar_put()
enviar_put()

despues_de_incrementar = leer_contador()
print(f"Después de incrementar 2 veces: {despues_de_incrementar}")

enviar_delete()

final = leer_contador()
print(f"Estado final: {final}")
